src/ParseRelation.js

```diff
@@ -50,12 +50,9 @@ class ParseRelation {
         if (this.parent.id !== parent.id) {
           throw new Error('Internal Error. Relation retrieved from two different Objects.');
         }
-      } else if (parent.id) {
```

Comment:
Do you know what as the purpose of this `parent.id` check?

Reply:
Not really. My educated guess is that it checks if the parent has been saved.

Reply:
I believe we'd need to understand this logic better and the purpose of the `parent.id` check before we remove it.

Reply:
I see. Unfortunately my team doesn't have the capacity for that, but we'd be more than happy to review/discuss any findings in this direction.